ghastly/ghastly.py

```diff
@@ -0,0 +1,315 @@
+                                     pebbles_left=pebbles_left)
+
+    main_filename = "pour_main_input.txt"
+    with open(main_filename, mode='w') as f:
+        f.write(main_text)
+
+    lmp = lammps()
+    lmp.file(main_filename)
+
+
+def pack_cyl(sim_block, element, rough_pf):
+    '''
+    Given a cylindrical core object, this function creates a corresponding
+    OpenMC region, and the the pack_spheres function to generate a list of
+    non-overlapping pebble centroids that fit within that region.
+
+    Parameters
+    ----------
+    sim_block : ghastly Sim object
+        Sim class object that contains simulation-specific simulation,
+        read from input_file.
+    element : ghastly CylCore object
+        CylCore object for a portion of the core model.
+    rough_pf : float
+        Packing fraction that OpenMC will pack this cylinder to.  Not the
+        same as the packing fraction in the input.
+
+    Returns
+    -------
+    coords : list
+        List of pebble centroid coordinates generated by OpenMC, within a
+        cylindrical region matching the core element passed to pack_cyl
+    '''
+    sides = openmc.ZCylinder(x0=element.x_c,
+                             y0=element.y_c,
+                             r=element.r)
+    top = openmc.ZPlane(z0=element.z_max)
+    bottom = openmc.ZPlane(z0=element.z_min)
+    region_bounds = -sides & -top & +bottom
+
+    coords = openmc.model.pack_spheres(sim_block.r_pebble,
+                                       region=region_bounds,
+                                       pf=rough_pf,
+                                       contraction_rate=sim_block.k_rate)
+
+    return list(coords)
+
+
+def find_box_bounds(sim_block, pour=False):
+    '''
+    Using the core_zones contained in the sim_block, this function
+    uses the dimensions of all core elements in the model to determine the
+    appropriate size of the bounding box used in LAMMPS simulations.
+    Note that when used to determine the bounding box for a simulation
+    using the pour LAMMPS fix, the bounding box is extended in the
+    positive z direction, to allow for a temporary insertion region to
+    be added.
+
+    Parameters
+    ----------
+    sim_block : Sim object
+        Sim object created from parameters in the input file for ghastly.
+    pour : bool
+        Default False.  Whether or not the LAMMPS simulation the bounds
+        are being generated for is using the pour fix in LAMMPS
+
+    Returns
+    -------
+    x_b : dict
+        Dictionary with key: value pairs giving the upper and lower
+        bounds in the x-direction for the bounding box.
+    y_b : dict
+        As x_b, but for the y direction.
+    z_b : dict
+        as x_b but for the z direction.
+    '''
+
+    core_list = (sim_block.core_intake |
+                 sim_block.core_main |
+                 sim_block.core_outtake)
+    x_list = []
+    y_list = []
+    z_list = []
+    for element in core_list.values():
```

Comment:
Create a geometry handling issue to simplify this kind of logic.

Ok for now as long as it's tested. 
Consider loading an external library (sympy? shapely? pyeuclid?)